build-system/tasks/visual-diff.js

```diff
@@ -35,10 +100,463 @@ function setPercyBranch() {
 }
 
 /**
- * Simple wrapper around the ruby based visual diff tests.
+ * Launches a background AMP webserver for unminified js using gulp.
+ *
+ * Waits until the server is up and reachable, and ties its lifecycle to this
+ * process's lifecycle.
+ *
+ * @return {!Promise} a Promise that resolves when the web server is launched
+ *     and reachable.
  */
-function visualDiff() {
-  setPercyBranch();
+async function launchWebServer() {
+  const gulpServeAsync = execScriptAsync(
+      `gulp serve --host ${HOST} --port ${PORT} ${process.env.WEBSERVER_QUIET}`,
+      {stdio: 'inherit'});
+
+  gulpServeAsync.on('exit', code => {
+    if (code != 0) {
+      log('error', colors.cyan("'serve'"),
+          `errored with code ${code}. Cannot continue with visual diff tests`);
+      process.exit(code);
+    }
+  });
+
+  process.on('exit', async() => {
+    if (gulpServeAsync.exitCode == null) {
+      gulpServeAsync.kill();
+      // The child node process has an asynchronous stdout. See #10409.
+      await sleep(100);
```

Comment:
This may not be needed. See my comment below, re: calling `process.exit`.

Reply:
Ack